crypto/asn1/a_d2i_fp.c

```diff
@@ -231,8 +233,10 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
     }
 
     *pb = b;
+    ERR_clear_last_mark();
     return off;
  err:
     BUF_MEM_free(b);
+    ERR_pop_to_mark();
```

Comment:
You must have confused the calls, 'cause it looks like you're throwing away all error reports on error...?